ssz-rs-derive/src/lib.rs

```diff
@@ -192,8 +192,9 @@ fn derive_deserialize_impl(data: &Data) -> TokenStream {
                     Some(field_name) => quote_spanned! { f.span() =>
                         let bytes_read = if <#field_type>::is_variable_size() {
                             let end = start + #BYTES_PER_LENGTH_OFFSET;
-                            let next_offset = u32::deserialize(&encoding[start..end])?;
-                            offsets.push((#i, next_offset as usize));
+                            let next_offset = u32::deserialize(&encoding[start..end])? as usize;
+                            offsets.last().map(|(_, previous_offset)| assert!(next_offset >= *previous_offset));
```

Comment:
this is a good point -- I think instead we should return an error here instead of panicking via the `assert!`

e.g. from the deserialization spec: https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md#deserialization

> Offsets: out of order, out of range, mismatching minimum element size.

let's either patch in this PR or ill make an issue to fix before v1

Reply:
Oh wow, I added an `assert!` 🙈 Sorry, definitely better to return an error here.